### PR TITLE
Add template rancher backup for cattle

### PR DESCRIPTION
**Pull request by disaster37**
It's a general purpose to solve backup matter on Rancher

This template deploy one container that manage backup aspect in your Rancher environment.
For the moment, this template permit to perform the following actions:
- Dump automatically all databases supported before start the remote backup:
  - PostgreSQL
  - MySQL
  - MariaDB
  - MongoDB
  - Elasticsearch
  - Dump automatically all rancher stacks settings (docker-compose.yml and rancher-compose.yml) before start the remote backup
  - Perform the remote backup of all `/backup` with duplicity (you can mount your shares storages on `/backup` to backup them)

We use the power of DIND (Docker in docker) to get the needed tools to perform dump, and then we use the power of duplicity to perform a remote Backup.
All auto discover work with the power of Rancher API

### CODE REVIEW

**Comment by smw355** [comment]
This looks really cool. Can't wait to see this added.

File: templates/rancher-backup/0/rancher-compose.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      default: "25"
+    - variable: RANCHER_API_URL
+      description: "The Rancher API URL with project ID"
+      label: "Rancher API URL"
+      type: "string"
+      required: true
+    - variable: RANCHER_API_KEY
+      description: "The Rancher API key"
+      label: "Rancher API key"
+      type: "string"
```
Comment (galal-hussein) [inline templates/rancher-backup/0/rancher-compose.yml]: @disaster37  Can you change this to type `password`

File: templates/rancher-backup/0/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  - MySQL
+  - MariaDB
+  - MongoDB
+  - Elasticsearch
+ - Dump automatically all rancher stacks settings (docker-compose.yml and rancher-compose.yml) before start the remote backup
+ - Perform the remote backup of all `/backup` with duplicity (you can mount your shares storages on `/backup` to backup them)
+
+ To manage your backup policy, you can use the following environment variables:
+ - `CRON_SCHEDULE`: when you should start backup (incremental if full is not needed). For example, to start backup each day set `0 0 0 * * *`
+ - `BACKEND`: this is the target URL to externalize the backup. For example, to use FTP as external backup set `ftp://login@my-ftp.com` and add environment variable `FTP_PASSWORD`. For Amazon S3, set `s3://host[:port]/bucket_name[/prefix]`. Read the ducplicity man for [all supported backend](http://duplicity.nongnu.org/duplicity.1.html#sect7). There are no default value.
```
Comment (galal-hussein) [inline templates/rancher-backup/0/README.md]: I couldn't find options for entering the AWS credential key to support the S3 backend.

File: templates/rancher-backup/0/rancher-compose.yml (hunk 1 above)
Comment (disaster37) [inline templates/rancher-backup/0/rancher-compose.yml]: I have modified this on my master branche

File: templates/rancher-backup/0/README.md (hunk 2 above)
Comment (disaster37) [inline templates/rancher-backup/0/README.md]: I have added 2 new entries to set the AWS key and AWS secret. Then you must use the backend url `s3://host[:port]/bucket_name[/prefix] `

(I have tested with minio, all works fine)

File: templates/rancher-backup/0/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    TARGET_PATH: ${TARGET_PATH}
+    BK_FULL_FREQ: ${BK_FULL_FREQ}
+    BK_KEEP_FULL: ${BK_KEEP_FULL}
+    BK_KEEP_FULL_CHAIN: ${BK_KEEP_FULL_CHAIN}
+    VOLUME_SIZE: ${VOLUME_SIZE}
+    RANCHER_API_URL: ${RANCHER_API_URL}
+    RANCHER_API_KEY: ${RANCHER_API_KEY}
+    RANCHER_API_SECRET: ${RANCHER_API_SECRET}
+    DEBUG: false
+    DISABLE_DUMP : false
```
Comment (disaster37) [inline templates/rancher-backup/0/docker-compose.yml]: I have added 2 new fileds for that

**Comment by rawmind0** [comment]
Hi @disaster37 ,

I apologize, but i rewrites history git repo by mistake. In order to be able to treat PR, it's needed that you fetch the rewritten history and rebase any local changes into the rewritten history.

I'm sorry, thank so much...

**Comment by galal-hussein** [comment]
closing in favor of https://github.com/rancher/community-catalog/pull/374

**Pull request closed without merging**